base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/views/ScriptGeneratorViewModel.java

```diff
@@ -284,24 +304,38 @@ public String getText(Object element) {
 		    }
 		};
 	}
+	
+	private PropertyChangeListener actionChangeListener = new PropertyChangeListener() {
+		
+		@Override
+		public void propertyChange(PropertyChangeEvent evt) {
+			actionChangeHandler(viewTable, btnGetValidityErrors, btnGenerateScript);
+		}
+	};
 
 	/**
-	 * Listen to changes on the actions and action validity property of the scriptGenerator table and
-	 *  update the view table.
-	 * 
-	 * @param table The view table to update.
-	 * @param btnGetValidityErrors The validity check button to style change.
-	 * @param btnGenerateScript The generate script button to style change.
+	 * The current viewTable in the actions.
 	 */
-	protected void bindValidityChecks(ActionsViewTable viewTable, Button btnGetValidityErrors, Button btnGenerateScript) {
-		this.scriptGeneratorModel.getScriptGeneratorTable().addPropertyChangeListener(ACTIONS_PROPERTY, e -> {
-			actionChangeHandler(viewTable, btnGetValidityErrors, btnGenerateScript);
-		});
-		this.scriptGeneratorModel.addPropertyChangeListener(VALIDITY_ERROR_MESSAGE_PROPERTY, e -> {
-			actionChangeHandler(viewTable, btnGetValidityErrors, btnGenerateScript);
-		});
-		// Listen for generated script refreshes
-		scriptGeneratorModel.addPropertyChangeListener(GENERATED_SCRIPT_FILEPATH_PROPERTY, evt -> {
+	private ActionsViewTable viewTable;
```

Comment:
Should be declared at the top before member functions are defined for readability purposes.

Reply:
Yes, changed.